bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryProcess.java

```diff
@@ -304,10 +335,38 @@ public void onServicesDiscovered() {
     public void onCharacteristicReadComplete(BluetoothCharacteristic characteristic, BluetoothCompletionStatus status) {
         serviceDiscoveryLock.lock();
         try {
-            if (characteristic.getGattCharacteristic() == GattCharacteristic.DEVICE_NAME) {
-                device.setName(characteristic.getStringValue(0));
-                nameDiscoveryCondition.signal();
+            if (status == BluetoothCompletionStatus.SUCCESS) {
+                switch (characteristic.getGattCharacteristic()) {
+                    case DEVICE_NAME:
+                        device.setName(characteristic.getStringValue(0));
+                        break;
+                    case MODEL_NUMBER_STRING:
+                        device.setModel(characteristic.getStringValue(0));
+                        break;
+                    case SERIAL_NUMBER_STRING:
+                        device.setSerialNumberl(characteristic.getStringValue(0));
+                        break;
+                    case HARDWARE_REVISION_STRING:
+                        device.setHardwareRevision(characteristic.getStringValue(0));
+                        break;
+                    case FIRMWARE_REVISION_STRING:
+                        device.setFirmwareRevision(characteristic.getStringValue(0));
+                        break;
+                    case SOFTWARE_REVISION_STRING:
+                        device.setSoftwareRevision(characteristic.getStringValue(0));
+                        break;
+                    default:
+                        break;
+                }
             }
+
+            ongoingDevInfoUUID.ifPresent(uuid -> {
+                if (uuid == characteristic.getGattCharacteristic().getUUID()) {
+                    ongoingDevInfoUUID = Optional.empty();
+                    infoDiscoveryCondition.signal();
+                }
+            });
```

Comment:
I think you can implement this without using Optional. Would be simpler to just use a GattCharacteristic instead of a UUID.

Reply:
Not sure if it's simpler, but not agains it either. I have understood that Optional should be used rather than nullable variables.

Reply:
I don't think there is anything wrong with nullable variables per se, just that it can become a pain to constantly do null checks when it would be reasonably expected to be non-null at the time.